### ISSUE TITLE
[BUG] Curseforge says its for retail????

### ISSUE THREAD

**Issue by CanadianJeff** [arkayenro/arkinventory#1553]
![Overwolf_jNcOzONRDq.png](https://i.imgur.com/mwhEkAN.png)

**Comment by arkayenro** [comment]
youd have to check with curseforge.  ai has three version, one for each client (and tagged that way)

**Comment by Arco-Dielhof** [comment]
Heyya! Volunteer CurseForge support moderator here.
I handled the issue from Jorvanus on Discord (I think that is @CanadianJeff that made the issue).
I checked the files from ArkInventory and it seems the names of your TOC files are not the standard expected names. Most likely not triggering the correct response from CurseForge. CurseForge uses the same names as Blizzard decided to use, so using those as well would be important to make things work properly. This would also explain why your users are saying this has been an issue since SL launch, but almost no other AddOns using multi-TOC support are having similar issues.

Your TOC files are named as follows:
Retail: Not present at all. Is supposed to be present and named: `ArkInventory-Mainline.toc`.
TBC: `ArkInventory_TBC` but is supposed to be named: `ArkInventory-BCC.toc`.
Classic: `ArkInventory_Vanilla` but is supposed to be named: `ArkInventory-Classic.toc`.

The `ArkInventory.toc` file you have, is the fall back, which is getting used if any of the proper files are missing, which means your TBC addon is loading that one, which you use for retail, thus flagging the AddOn as a 'retail' addon.

You should be easily able to fix the issue for your users if you just rename your files to the correct names and make sure that the fallback TOC file is not specifically used for retail, but there is an actual retail TOC file present.

Let me know if you have any questions or if I can assist with anything else :D

**Comment by Arco-Dielhof** [comment]
Seemingly there is more than one issue at hand here. I did some further digging to be sure and had some help from OW themselves.

You are uploading 3 different files, but trying to use Multi-TOC support like you are uploading 1 file for all 3 game versions. These are two different systems and you can not use half of each of them.

You need to either use 1 upload, with a single TOC files. Or 3 uploads, with each 1 TOC file.

**Comment by arkayenro** [comment]
@Arco-Dielhof  those formats are old, you might want to check https://wowpedia.fandom.com/wiki/TOC_format and https://github.com/Stanzilla/WoWUIBugs/issues/68#issuecomment-889431675

i used to upload only a single file to curseforge and just tag it there for all three clients (as my mod is coded to work across all clients) but that failed miserably so i had to resort to multi-toc files (so the users got the right toc version) and multiple uploads (to stop curse from failing)

can you confirm curseforge/overwolf will now actually support a single upload with multi-toc files in it?

**Comment by CanadianJeff** [comment]
oh and for the record I just found out about https://wowup.io/ so I am probley just gonna dump curseforge because of the overwolf bloat that comes with it

**Comment by tomwolfgang** [comment]
@CanadianJeff Hey - I'm from CF.
What bloat? I see that on disk, WoWup takes up 370MB while Overwolf 273MB.

Overwolf is a platform similar to electron - it enables app development (CF being one).
As a user of CF, you aren't forced into any other app. If, in the future, you'd like to use another OW-based app (e.g. Warcraft Log Companion) - it'll just save you from having to install another 270-370MB program.

**Comment by tomwolfgang** [comment]
@arkayenro yes - CF supports one addon tagged by all three (for a while now).
Having no Mainline should work, as long as the default toc is for retail.

The game should fallback to the relevant toc file.

**Comment by CanadianJeff** [comment]
sorry i'm not talking about storage space bloat heck I have a 4TB C:\ drive i'm talking about pointless 3rd party apps

**Comment by arkayenro** [comment]
@tomwolfgang > but not the packager.

well thats how ive uploaded since day one.  tag the svn on curse and the packager does its thing

im trying the bigwigspackager instead but that seems to be just as problematical, although i presume once i get it setup right it can send one file to curse and tag it for all three clients

**Comment by tomwolfgang** [comment]
@CanadianJeff I understand, obviously apps like porofessor.gg aren't interesting for non League players.
Keep in mind that they are not installed/downloaded without having to explicitly download/install them.

However, if you do wish to (explicitly) install an app, like Warcraft Logs Companion or WOWTBC, it will utilize the same engine and save you time (for installing) and space.

**Comment by arkayenro** [comment]
> If you are up for it, you could create the package and then later on tag it via

i forked the bigwigs packager code and rewrote it to handle uploading a single file, with multiple client based tocs, and tagged based of the interface version in those tocs

it seems to be working ok with my test mod but curse has some issues that i need to work around

**Comment by arkayenro** [comment]
@tomwolfgang just two so far and i've opened tickets for them

1. Archive is missing ToC files (name is case sensitive): ArkTest/ArkTest.toc.

my test mod (https://www.curseforge.com/wow/addons/arktest) normally has an ArkTest-Mainline.toc file, it has no default toc as i wanted to test the packager code could handle an addon that only worked in one game client thus didnt have a default toc, only a game specific one.

curse currently doesnt like that and while it will initially accept the release theres some (automated) checking done later that notices thers no toc file and rejects the upload.

2) cosmetic - when you upload a file tagged for all three game clients curse appears to pick one at random (although its consistent about it) to use as the "primary" game version.

i sorted the version ids (so that retail was either first or last) in the metadata for the upload but that didnt make any difference, it keeps picking the classic version (it did pick the tcc version for a while)

i thought it might have been due to having a mainline toc instead of a default but it still does it after i put one back in

**Comment by CanadianJeff** [comment]
just check this issue has been solved CLOSING
